### ISSUE TITLE
Scraping from FIFA doesn't work for 2018 games

### ISSUE THREAD

**Issue by siltus** [neaorin/PredictTheWorldCup#1]
Hey,
Not sure you're still working on this but I played with your tutorial (real informative and helpful, thanks!). I noticed that acquiring game results from FIFA using the current scripts stopped working. Instead I was able to download game scores in a JSON format, and I also wrote a script to transform these JSON responses to the format the `fifa_transformcsv.js` script can work with (and from that point, it just works the same as you wrote it).
Feel free using it if you plan on updating your code...

**Download JSONs**
```bash
#!/bin/bash

# based on traffic to/from : http://www.fifa.com/live-scores/international-tournaments/fixtures-results/index.html

for idx in `seq 1980 2018`;
do
	if [ ! -f "$idx.json" ]; then
		DL_LINK="http://data.fifa.com/livescores/en/internationaltournaments/matches/m/bydaterange/$idx-1-01/$((idx + 1))-1-01"
		curl -f $DL_LINK -o $idx.json
		sed -i.bak -e "s/^_matchesByDateRangeCallback(//" -e "s/)$//" $idx.json
		rm -rf *.bak
	fi
done
```

**Transform for your scripts**
```javascript
var glob = require("glob")
var fs = require("fs")
var moment = require("moment")

// keep skippedTournaments lowercase!!!
const skippedTournaments = [ 'beach', 'futsal', 'u17', 'u-17', 'u-20', 'friendly', 'friendlies', 'club world cup', 'youth olympic' ]

var processedMatches = []

let processMatchFile = function(filename) {
	var contents = fs.readFileSync(filename);
 	var jsonContent = JSON.parse(contents);

 	for (competitionId in jsonContent.competitionslist) {
 		const competition = jsonContent.competitionslist[competitionId];

 		var shouldSkip = false;

 		// skip non-official grass football games
 		for (idx in skippedTournaments) {
	 		if (competition.name.toLowerCase().indexOf(skippedTournaments[idx]) != -1) { 
	 			shouldSkip = true;
	 			break;
	 		}

	 		if (competition.competitionSeoName.toLowerCase().indexOf(skippedTournaments[idx]) != -1) { 
	 			shouldSkip = true;
	 			break;
	 		}

 		}

 		// skip club competitions
 		if (!shouldSkip && competition.isClubCompetition === true) {
 			shouldSkip = true;
 		}

 		if (shouldSkip) {
 			console.log(`Skipping ${competition.name} [${competition.competitionSeoName}] ...`);
 			continue;
 		}

		console.log(`Processing ${competition.name} [${competition.competitionSeoName}] ...`);
		processTournament(competition)
 	}
}

let processTournament = function(tournamentJson) {
	for (gameJsonId in tournamentJson.matchlist) {
		gameJson = tournamentJson.matchlist[gameJsonId]

		if (!gameJson.isFinished) {
			//console.log(`... Skipping ${gameJson.homeTeamName} vs ${gameJson.awayTeamName} on ${gameJson.matchDateUTC} - not finished`)
			continue;
		}

		//console.log(`... Processing ${gameJson.homeTeamName} vs ${gameJson.awayTeamName} on ${gameJson.matchDateUTC}`)

		var rawDate = gameJson.matchDate

		if (gameJson.hasOwnProperty('matchDateUTC')) {
			rawDate = gameJson.matchDateUTC
		}

		var date = moment(rawDate);
		var dateComponent = date.utc().format('YYYYMMDD');

		// console.log(`${rawDate} ==> ${date} ==> ${dateComponent}`);

		var venue = ''
		if (gameJson.hasOwnProperty('venue')) {
			var venue = gameJson.venue.trim()
		} 

		var processedMatch = {
	      "date": dateComponent,
	      "team1": gameJson.homeCountryCode,
	      "team1Text": gameJson.homeTeamName,
	      "team2": gameJson.awayCountryCode,
	      "team2Text": gameJson.awayTeamName,
	      "venue": venue,
	      "IdCupSeason": gameJson.idCupSeason.toString(),
	      "CupName": gameJson.competitionName,
	      "team1Score": gameJson.scoreHome,
	      "team2Score": gameJson.scoreAway,		
		}

		// reasonWinCode values from: http://js.fifa.com/components/script/require-libs/lang=en/main.js
		// for our purposes only 3 is important enough

		// "fifa.winningReason_0":{text:"Draw", abbr:"Draw"},
		// "fifa.winningReason_2":{text:"{WinTeamName} win after extra time", abbr:"{WinTeamName} win AET"},
		// "fifa.winningReason_3":{text:"{WinTeamName} win on penalties ({ScorePenH} - {ScorePenA})", abbr:"{WinTeamCountry} win after PSO"},
		// "fifa.winningReason_4":{text:"{WinTeamName} win on aggregate after regular time ({ScoreAggH} - {ScoreAggA})", abbr:"{WinTeamCountry} Win on agg."},
		// "fifa.winningReason_5":{text:"{WinTeamName} win on aggregate after extra time ({ScoreAggH} - {ScoreAggA})", abbr:"{WinTeamCountry} win on agg AET"},
		// "fifa.winningReason_6":{text:"{WinTeamName} win on away goal after regular time", abbr:"{WinTeamCountry} win on away goals"},
		// "fifa.winningReason_7":{text:"{WinTeamName} win on away goal after extra time", abbr:"{WinTeamCountry} win on away goals AET"},
		// "fifa.winningReason_8":{text:"Win by Silver Goal", abbr:"Win by Silver Goal"},
		// "fifa.winningReason_9":{text:"Win on Golden Goal", abbr:"Win by Golden Goal"},

		if (gameJson.hasOwnProperty('reasonWinCode')) {
			if (gameJson.reasonWinCode === 3) {
				processedMatch["team1PenScore"] = gameJson.scorePenaltyHome
				processedMatch["team2PenScore"] = gameJson.scorePenaltyAway
				processedMatch["statText"] = "Win on penalty"
				processedMatch["resText"] = `${gameJson.scoreHome}-${gameJson.scoreAway} (${gameJson.scorePenaltyHome}-${gameJson.scorePenaltyAway})`				
			}

		} else {
			processedMatch["statText"] = ""
			processedMatch["resText"] = `${gameJson.scoreHome}-${gameJson.scoreAway}`
		}

		processedMatches.push(processedMatch)
		//console.log(gameJson)
	}
}

glob("[1-2]*.json", function (er, files) {

	files.forEach(function(file) {
    	 processMatchFile(file);
	});

	fs.writeFileSync('matches.json', JSON.stringify(processedMatches));

	console.log('done')
})
```

**Comment by ruimgbarros** [comment]
@siltus, very useful! I was able to download the data on the JSON format, but I'm not really sure how should I run the second part of the code...

**Comment by neaorin** [comment]
@siltus thanks! I did notice a few weeks ago that my script didn't work on newer games, but I never got around to figuring out why. I'll take a look and update the code.

**Comment by claucarrera** [comment]
@neaorin  Thank you so much for this Tutorial, I just love it.  
@siltus     Thank you for your solution in order to bring newer games :)

**Comment by antjefitzner** [comment]
Getting the file worked fine! How do I run the second script, @siltus ? TIA

**Comment by soulstar205** [comment]
@neaorin, I was hoping you could post results here. I mean the predictions, I'm having serious computer issues now and will not be able to run the codes. Don't wannaiss out on this

**Comment by azzouzze** [comment]
it did not work for me neither. though the URL seems fine, the saving of the files gets 0 bites.

**Comment by neaorin** [comment]
Fixed by commit [fea7753156b9ce7b59c74c4acc12a761e21db29f](https://github.com/neaorin/PredictTheWorldCup/commit/fea7753156b9ce7b59c74c4acc12a761e21db29f)

[README file](https://github.com/neaorin/PredictTheWorldCup/blob/master/acquisition/fifa/README.md) has updated steps.

@siltus thank you very much for your help in resolving this issue.
